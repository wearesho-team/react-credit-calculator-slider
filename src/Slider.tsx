import * as React from "react";
import * as Calculator from "react-credit-calculator";
import BaseSlider, { SliderProps } from "rc-slider";

export const Slider: React.FunctionComponent<SliderProps> = (props) => (
    <Calculator.ControlContext.Consumer>
        {(value: Calculator.ControlContextValue) => <BaseSlider {...props} {...value} />}
    </Calculator.ControlContext.Consumer>
);
