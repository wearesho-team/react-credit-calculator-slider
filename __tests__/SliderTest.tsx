import * as React from "react";
import * as Calculator from "react-credit-calculator";
import * as renderer from "react-test-renderer";

import { Slider } from "../src";

describe("<Slider />", () => {
    const value: Calculator.ControlContextValue = {
        min: 250,
        max: 2000,
        value: 1000,
        step: 50,
        onChange: (next) => next,
    };

    it("renders correctly", () => {
        const tree = renderer
            .create(
                <Calculator.ControlContext.Provider value={value}>
                    <Slider/>
                </Calculator.ControlContext.Provider>
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
