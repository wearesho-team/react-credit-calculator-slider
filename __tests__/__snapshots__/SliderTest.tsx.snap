// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Slider /> renders correctly 1`] = `
<div
  className="rc-slider"
  onBlur={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onMouseDown={[Function]}
  onMouseUp={[Function]}
  onTouchStart={[Function]}
  style={undefined}
>
  <div
    className="rc-slider-rail"
    style={Object {}}
  />
  <div
    className="rc-slider-track"
    style={
      Object {
        "left": "0%",
        "width": "42.857142857142854%",
      }
    }
  />
  <div
    className="rc-slider-step"
  />
  <div
    aria-disabled={false}
    aria-valuemax={2000}
    aria-valuemin={250}
    aria-valuenow={1000}
    className="rc-slider-handle"
    onBlur={[Function]}
    onKeyDown={[Function]}
    role="slider"
    style={
      Object {
        "left": "42.857142857142854%",
      }
    }
    tabIndex={0}
  />
  <div
    className="rc-slider-mark"
  />
</div>
`;
